packages/registry/src/utils/get-comp-with-state.tsx

```diff
@@ -3,43 +3,27 @@ import {ISimulation} from '../registry';
 
 export function getCompWithState(Comp: React.ComponentClass<any>, simulation: ISimulation<any, any>): JSX.Element {
   // Override the requested component's setState
-  manageComponent(Comp.prototype, (state: any) => {
+  const getOverrideState = (state: any) => {
     return {
       ...state,
       ...simulation.state
     };
-  });
+  };
 
   class Wrapped extends Comp {
     constructor(props: any) {
       super(props);
-      // Save the old state
-      const oldState = super.state;
-      // Replace the old state with the old state + the override state
-      super.state = {...oldState, ...simulation.state};
-    }
-
-    public render() {
-      // Just return the extended Component's render method
-      return super.render();
+      this.state = {...this.state, ...simulation.state};
```

Comment:
```
class Wrapped extends Comp {
  constructor() {
    ...
  }

  setState(state) {
    super.setState(getOverrideState(state));
  }
}
```

Not sure if this is important at this point, but `setState()` can also receive an updater function and a callback. And there's also `getDerivedStateFromProps` which can affect the state.